platforms/Windows/Directory.Build.props

```diff
@@ -24,34 +29,58 @@
   <PropertyGroup>
     <RootBuildFolder>$([System.IO.Path]::GetFullPath($(MSBuildThisFileDirectory)build\))</RootBuildFolder>
     <BaseOutputPath>$(RootBuildFolder)</BaseOutputPath>
+    <BaseOutputPath Condition="!HasTrailingSlash('$(BaseOutputPath)')">$(BaseOutputPath)\</BaseOutputPath>
     <BaseIntermediateOutputPath>$(BaseOutputPath)obj\$(MSBuildProjectName)\</BaseIntermediateOutputPath>
     <OutputPath>$(BaseOutputPath)$(Configuration)\$(ProductArchitecture)\</OutputPath>
 
-    <DefaultCompressionLevel Condition="'$(DefaultCompressionLevel)' == ''">high</DefaultCompressionLevel>
+    <BundleFlavor Condition="'$(BundleFlavor)' == ''">web</BundleFlavor>
   </PropertyGroup>
 
   <PropertyGroup>
-    <PackageScope>perMachine</PackageScope>
+    <!--
+    ICE38 is about mixing per-user and per-machine resources (not a thing for us).
+    ICE61 is a warning about allowing "same-version" major upgrades, something we want.
+    ICE64 is documented as not being an issue when packages are always per-user.
+    ICE91 is about "roaming scenarios," which doesn't apply to our use of LocalAppDataFolder.
+    -->
+    <SuppressIces>ICE38;ICE61;ICE64;ICE91</SuppressIces>
+    <PackageScope>perUser</PackageScope>
+  </PropertyGroup>
+
+  <!-- High package and high bundle compression results in the smallest bundle, at the cost of build time. -->
+  <PropertyGroup Condition="'$(BundleFlavor)' == 'offline'">
+    <PackageCompressionLevel>high</PackageCompressionLevel>
+    <ArePackageCabsEmbedded>false</ArePackageCabsEmbedded>
+
     <IsBundleCompressed>true</IsBundleCompressed>
+    <BundleCompressionLevel>high</BundleCompressionLevel>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(BundleFlavor)' == 'web'">
+    <PackageCompressionLevel>high</PackageCompressionLevel>
+    <ArePackageCabsEmbedded>false</ArePackageCabsEmbedded>
```

Comment:
Since we do not package the cabs, I assume that there needs to signing for the cabs?  How do we set that up into the pipline?  (Currently, `-p:PASSPHRASE=... -p:CERTIFICATE=... -p:SignOutput=true` would sign the MSI and bootstrapper).

Reply:
Actually, there's no security benefit from signing .cabs. Burn stores a SHA-512 hash of every payload in the bundle manifest (which is protected by the signature on the bundle .exe itself). But WiXCodeSigning.targets already has the SignCabs target, so it might already be happening.

Reply:
Ah, okay.  I guess I am also curious why not embed the cabs in the MSI?  Does it help reduce the download size somehow?

Reply:
Burn caches packages so it can always uninstall and repair. MSI also caches packages for uninstall. If the cab is embedded, you have two copies and MSI doesn't always use its cached copy as a source for repair. With an external .cab, MSI caches only the tiny .msi file and not the .cab.

Reply:
Ah!  This would be great to add to the docs!

Reply:
Done.